### PR TITLE
Certificate revocation

### PR DESCRIPTION
**Pull request by spivachuk**
- Added unit tests of Handler for certificates revocation.
- Corrected and enhanced unit tests of Handler for certificates addition.
- Fixed bugs in the certificates revocation feature.
- Enhanced REST API and CLI integration test​s of PKI for certificates revocation.
- Fixed an issue with wrong matches between requests and endpoints in REST API for PKI.

### CODE REVIEW

File: x/pki/handler_test.go [hunk 1]
```diff
@@ -404,107 +489,704 @@ func TestHandler_AddX509Cert_ForFailedCertificateVerification(t *testing.T) {
...
+
 	setup.PkiKeeper.AddApprovedCertificate(setup.Ctx, rootCert)
+	setup.PkiKeeper.SetUniqueCertificateKey(setup.Ctx, rootCert.Subject, rootCert.SerialNumber)
 
 	// store second root certificate
 	rootCert = rootCertificate(setup.Trustee)
 	rootCert.SerialNumber = SerialNumber
+
 	setup.PkiKeeper.AddApprovedCertificate(setup.Ctx, rootCert)
+	setup.PkiKeeper.SetUniqueCertificateKey(setup.Ctx, rootCert.Subject, rootCert.SerialNumber)
```
Comment (askolesov) [inline x/pki/handler_test.go:571]: Shouldn't it be an internal logic of the keeper?

File: x/pki/handler_test.go [hunk 2]
```diff
@@ -404,107 +489,704 @@ func TestHandler_AddX509Cert_ForFailedCertificateVerification(t *testing.T) {
...
+	// check that approved certificate still exist
+	certificate, _ := querySingleCertificate(&setup, constants.RootSubject, constants.RootSubjectKeyID)
+	require.NotNil(t, certificate)
+
+	// check that revoked certificate does not exist
+	_, err := queryRevokedCertificates(&setup, constants.RootSubject, constants.RootSubjectKeyID)
+	require.Equal(t, types.CodeRevokedCertificateDoesNotExist, err.Code())
+
+	// check that unique certificat key stays registered
+	setup.PkiKeeper.IsUniqueCertificateKeyPresent(setup.Ctx, constants.RootIssuer, constants.RootSerialNumber)
```
Comment (askolesov) [inline x/pki/handler_test.go:835]: Maybe require true?

File: x/pki/handler_test.go [hunk 3]
```diff
@@ -404,107 +489,704 @@ func TestHandler_AddX509Cert_ForFailedCertificateVerification(t *testing.T) {
...
+	// check that approved certificate does not exist anymore
+	_, err = queryCertificates(&setup, constants.RootSubject, constants.RootSubjectKeyID)
+	require.Equal(t, types.CodeCertificateDoesNotExist, err.Code())
+
+	// query and check revoked certificate
+	revokedCertificate, _ := querySingleRevokedCertificate(&setup, constants.RootSubject, constants.RootSubjectKeyID)
+	require.Equal(t, certificateBeforeRevocation, revokedCertificate)
+
+	// check that unique certificate key stays registered
+	setup.PkiKeeper.IsUniqueCertificateKeyPresent(setup.Ctx, constants.RootIssuer, constants.RootSerialNumber)
```
Comment (askolesov) [inline x/pki/handler_test.go:877]: Same is here

**Review by askolesov — COMMENTED** [review]
Great tests!

File: x/pki/handler_test.go (hunk 2 above)
Comment (spivachuk) [reply x/pki/handler_test.go]: Fixed.

File: x/pki/handler_test.go (hunk 3 above)
Comment (spivachuk) [reply x/pki/handler_test.go]: Fixed.

File: x/pki/handler_test.go (hunk 1 above)
Comment (spivachuk) [reply x/pki/handler_test.go:572]: Created a separate ticket for this: https://github.com/zigbee-alliance/distributed-compliance-ledger/issues/51

**Review by askolesov — APPROVED** [review]
👍

**Pull request merged**